packages/core/strapi/tests/api/relations.test.e2e.js

```diff
@@ -0,0 +1,690 @@
+    method: 'PUT',
+    url: `/api/${pluralName}/${id}`,
+    body: { data },
+    qs: { populate },
+  });
+  return body.data;
+};
+
+describe('Relations', () => {
+  const builder = createTestBuilder();
+
+  beforeAll(async () => {
+    await builder.addComponent(compo(false)).addContentTypes([productModel, shopModel]).build();
+
+    await modelsUtils.modifyComponent(compo(true));
+
+    strapi = await createStrapiInstance();
+    rq = await createAuthRequest({ strapi });
+
+    const createdProduct1 = await createEntry('products', { name: 'Skate' });
+    const createdProduct2 = await createEntry('products', { name: 'Candle' });
+    const createdProduct3 = await createEntry('products', { name: 'Mug' });
+
+    data.products.push(createdProduct1);
+    data.products.push(createdProduct2);
+    data.products.push(createdProduct3);
+
+    id1 = data.products[0].id;
+    id2 = data.products[1].id;
+    id3 = data.products[2].id;
+  });
+
+  afterAll(async () => {
+    await strapi.destroy();
+    await builder.cleanup();
+  });
+
+  describe.each([['connect'], ['set']])(
+    'Create an entity with relations using %s',
+    (connectOrSet) => {
+      describe.each([
+        ['directly in the array ([1, 2])', 'object'],
+        ['an object in the array ([{ id: 1 }, { id: 2 }])', 'array'],
+      ])('ids being %s', (name, mode) => {
+        test('In one order', async () => {
+          const oneRelation = mode === 'object' ? [{ id: id1 }] : [id1];
+          const manyRelations = mode === 'object' ? [{ id: id1 }, { id: id2 }] : [id1, id2];
+
+          const shop = await createEntry(
+            'shops',
+            {
+              name: 'Cazotte Shop',
+              products_ow: { [connectOrSet]: oneRelation },
+              products_oo: { [connectOrSet]: oneRelation },
+              products_mo: { [connectOrSet]: oneRelation },
+              products_om: { [connectOrSet]: manyRelations },
+              products_mm: { [connectOrSet]: manyRelations },
+              products_mw: { [connectOrSet]: manyRelations },
+              myCompo: {
+                compo_products_ow: { [connectOrSet]: oneRelation },
+                compo_products_mw: { [connectOrSet]: manyRelations },
+              },
+            },
+            populateShop
+          );
+
+          expect(shop).toMatchObject({
+            attributes: {
+              myCompo: {
+                compo_products_mw: { data: [{ id: id1 }, { id: id2 }] },
+                compo_products_ow: { data: { id: id1 } },
+              },
+              products_mm: { data: [{ id: id1 }, { id: id2 }] },
+              products_mo: { data: { id: id1 } },
+              products_mw: { data: [{ id: id1 }, { id: id2 }] },
+              products_om: { data: [{ id: id1 }, { id: id2 }] },
+              products_oo: { data: { id: id1 } },
+              products_ow: { data: { id: id1 } },
+            },
+          });
+        });
+
+        test('In reversed order', async () => {
```

Comment:
curious, why do you check inserting the ids in the reverse order?

Reply:
To ensure that having the right order in the test above was not by chance :p